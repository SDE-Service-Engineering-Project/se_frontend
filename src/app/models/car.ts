export interface Car {
  carId: number;
  createdOn: string;
  brand: string;
  model: string;
  constructionYear: number;
  price: number;
  precision: number;
  currency: string;
}
