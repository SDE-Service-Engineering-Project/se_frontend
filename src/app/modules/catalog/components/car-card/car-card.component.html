<div class="card" style="width: 18rem">
  <img
    src="https://pictures.dealer.com/c/costamesacadillac/0679/a976281f56ff06d46dc390283aa14da6x.jpg?impolicy=downsize_bkpt&imdensity=1&w=520"
    class="card-img-top"
    alt="{{ car.brand }} {{ car.model }}"
  />
  <div class="card-body">
    <h5 class="card-title">{{ car.brand }} {{ car.model }}</h5>
    <p class="card-text">Brand: {{ car.brand }}</p>
    <p class="card-text">Model: {{ car.model }}</p>
    <p class="card-text">Manufacture Year: {{ car.constructionYear }}</p>
    <p class="card-text"><b>Price:</b> {{ car.price | currency: car.currency }} / Day</p>
    <button (click)="selectCar()" style="float: right" class="btn btn-primary">
      Rent
    </button>
  </div>
</div>
