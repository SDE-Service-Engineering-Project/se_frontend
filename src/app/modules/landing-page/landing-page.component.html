<div> Welcome to Car Rental Service</div>
<app-maps></app-maps>
