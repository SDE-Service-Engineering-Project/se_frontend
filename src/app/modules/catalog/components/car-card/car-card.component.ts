import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Car } from '../../../../models/car';

@Component({
  selector: 'app-car-card',
  templateUrl: './car-card.component.html',
  styleUrls: ['./car-card.component.sass'],
})
export class CarCardComponent {
  @Input() car: Car = {} as Car;
  @Output() carSelected: EventEmitter<Car> = new EventEmitter<Car>();

  selectCar() {
    this.carSelected.emit(this.car);
  }
}
