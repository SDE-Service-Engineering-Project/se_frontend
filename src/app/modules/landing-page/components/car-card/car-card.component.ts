import {Component, EventEmitter, Input, Output} from '@angular/core';
import {Car} from "../../../../models/car";

@Component({
  selector: 'app-car-card',
  templateUrl: './car-card.component.html',
  styleUrls: ['./car-card.component.sass']
})
export class CarCardComponent {
  @Input() car: Car = {} as Car;
  @Output() carRented: EventEmitter<Car> = new EventEmitter<Car>()

  rentCar() {
    this.carRented.emit(this.car);
  }

}
