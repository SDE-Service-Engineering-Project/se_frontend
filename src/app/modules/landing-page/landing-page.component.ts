import { Component } from '@angular/core';
import {carMock1, carsMock} from '../../utils/testing/mocks/car.mock';
import { Car } from '../../models/car';

@Component({
  selector: 'app-landing-page',
  templateUrl: './landing-page.component.html',
  styleUrls: ['./landing-page.component.sass']
})
export class LandingPageComponent {
  cars: Car[] = carsMock;

  rentCar(car: Car) {
    console.log(car);
  }
}
