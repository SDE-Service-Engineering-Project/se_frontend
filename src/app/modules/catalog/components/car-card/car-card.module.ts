import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CarCardComponent } from './car-card.component';

@NgModule({
  imports: [CommonModule],
  declarations: [CarCardComponent],
  exports: [CarCardComponent],
})
export class CarCardModule {}
