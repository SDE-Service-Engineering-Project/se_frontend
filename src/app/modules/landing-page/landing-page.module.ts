import { NgModule } from '@angular/core';
import { LandingPageRoutingModule } from './config/landing-page.routing';
import { LandingPageComponent } from './landing-page.component';
import { CommonModule } from '@angular/common';
import { CarCardComponent } from './components/car-card/car-card.component';

@NgModule({
  imports: [LandingPageRoutingModule, CommonModule],
  declarations: [LandingPageComponent, CarCardComponent],
})
export class LandingPageModule {}
