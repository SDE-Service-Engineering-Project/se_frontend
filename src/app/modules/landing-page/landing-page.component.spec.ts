import { ComponentFixture } from '@angular/core/testing';

import { LandingPageComponent } from './landing-page.component';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { carsMock } from '../../utils/testing/mocks/car.mock';
import { CarCardComponent } from './components/car-card/car-card.component';
import { CarCardModule } from './components/car-card/car-card.module';

describe('LandingPageComponent', () => {
  let component: LandingPageComponent;
  let fixture: ComponentFixture<LandingPageComponent>;
  let spectator: Spectator<LandingPageComponent>;

  const createComponent = createComponentFactory({
    component: LandingPageComponent,
    imports: [CarCardModule],
  });

  beforeEach(() => {
    spectator = createComponent();
    fixture = spectator.fixture;
    component = spectator.component;
  });

  beforeEach(() => {
    component.cars = carsMock;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
