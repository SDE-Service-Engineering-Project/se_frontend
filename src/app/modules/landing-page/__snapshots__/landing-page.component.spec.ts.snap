// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageComponent should render the component 1`] = `
<div
  id="root0"
>
  <div
    class="container-fluid mt-4"
  >
    <div
      class="d-flex flex-column align-items-center justify-content-center"
    >
      <div
        class="row mb-4"
      >
        <h1>
          Welcome to Car Booking Service
        </h1>
      </div>
      <div
        class="row"
        style="min-width: 100%;"
      >
        <app-maps>
          <google-map
            height="300px"
            width="100%"
          />
        </app-maps>
      </div>
      <div
        class="row mb-4 mt-4"
      >
        <h2>
          Select your preferred time range
        </h2>
      </div>
      <div
        class="row"
      >
        <div
          class="col-12 d-flex justify-content-center"
        >
          <app-date-picker>
            <form
              class="row row-cols-sm-auto"
            >
              <div
                class="col-12"
              >
                <label
                  class="form-label"
                  for="pick-up-date"
                >
                  Pick-up Date
                </label>
                <div
                  class="dp-hidden position-absolute"
                >
                  <div
                    class="input-group"
                  >
                    <input
                      class="form-control"
                      name="datepicker"
                      ngbdatepicker=""
                      outsidedays="hidden"
                      tabindex="-1"
                    />
                    
                    
                  </div>
                </div>
                <div
                  class="input-group"
                >
                  <input
                    class="form-control"
                    id="pick-up-date"
                    name="dpFromDate"
                    placeholder="yyyy-mm-dd"
                  />
                  <button
                    class="btn btn-outline-secondary bi bi-calendar3"
                    data-testid="from-date-btn"
                    type="button"
                  />
                </div>
              </div>
              <div
                class="col-12"
              >
                <label
                  class="form-label"
                  for="return-date"
                >
                  Return Date
                </label>
                <div
                  class="input-group"
                >
                  <input
                    class="form-control"
                    id="return-date"
                    name="dpToDate"
                    placeholder="yyyy-mm-dd"
                  />
                  <button
                    class="btn btn-outline-secondary bi bi-calendar3"
                    data-testid="to-date-btn"
                    type="button"
                  />
                </div>
              </div>
            </form>
          </app-date-picker>
        </div>
        <div
          class="col-12 mt-4 d-flex justify-content-center"
        >
          <button
            class="btn btn-primary"
            data-testid="search-btn"
            id="search-btn"
            style="min-width: 300px;"
            type="submit"
          >
            Show available cars
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
